src/main/java/plugily/projects/buildbattle/handlers/menu/OptionsMenuHandler.java

```diff
@@ -37,69 +36,59 @@
  * <p>
  * Created at 23.12.2018
  */
-public class OptionsMenuHandler implements Listener {
-
-  private final Main plugin;
+public final class OptionsMenuHandler {
 
   public OptionsMenuHandler(Main plugin) {
-    this.plugin = plugin;
-    plugin.getServer().getPluginManager().registerEvents(this, plugin);
-  }
-
-  @EventHandler
-  public void onOptionsMenuClick(InventoryClickEvent event) {
-    HumanEntity humanEntity = event.getWhoClicked();
-
-    if(!(humanEntity instanceof Player)) {
-      return;
-    }
-
-    if(event.getInventory() != plugin.getOptionsRegistry().formatInventory()) {
-      return;
-    }
-
-    BaseArena arena = plugin.getArenaRegistry().getArena((Player) humanEntity);
-    if(arena == null || arena.getArenaState() != ArenaState.IN_GAME) {
-      return;
-    }
-
-    ItemStack currentItem = event.getCurrentItem();
-
-    for(MenuOption option : plugin.getOptionsRegistry().getRegisteredOptions()) {
-      if(!option.getItemStack().isSimilar(currentItem)) {
-        continue;
+    plugin.getServer().getPluginManager().registerEvent(InventoryClickEvent.class, new Listener() {
+    }, org.bukkit.event.EventPriority.NORMAL, new org.bukkit.plugin.EventExecutor() {
```

Comment:
importing EventPriortiy and EventExecutor looks fine for me to have better readability ;)